torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -2899,15 +2903,24 @@ c10::intrusive_ptr<Work> ProcessGroupNCCL::pointToPoint(
   bool isSendRecvSelf = false;
   // For batch_isend_irecv, ncclGroupStart() would be called upfront
   bool batchP2P = ncclActiveGroupCounter_ > 0;
-  if (batchP2P) {
-    // For batch P2P, we need to treat it like a collective when selecting
-    // communicator, because other ranks can call into this batch other than my
-    // rank and my peer
-    key = getKeyFromDevice(device);
-    p2pRank = rank_;
-    p2pTargetRank = peer;
-  } else {
-    // For single P2P, preserve the old two-rank behavior (to avoid perf diff)
+
+  // If an existing communicator exists, we want to use it (for batch or
+  // non-batch  P2P)
+  key = getKeyFromDevice(device);
+  p2pRank = rank_;
+  p2pTargetRank = peer;
```

Comment:
Nit:
combine these with the declarations in Lines 2901 and 2902 for 2 less lines of code.
```
auto key = getKeyFromDevice(device);
auto p2pRank = rank_;
auto p2pTargetRank = peer;
```